### ISSUE TITLE
Wrongly error throwing

### ISSUE THREAD

**Issue by diegoaguilar** [jshint/jshint#2844]
I got a line with a regex (ugly code, I know) which made JSHint throw an error when I think it shouldn't have:

So this is the line:

```
post.form.email = `demo+${/(.*)@.+/.exec(req.body.email)[1]}@lumahealth.io`;
```

Which basically sets `post.form.email` to the result of concatenating `'demo+'` to the first capture from a regex applied on `req.body.email` and also concatenates `'@lumahealth.io'`, all this with ES6 templates.

This code actually works, I tested it but running JSHint against it will throw:

![image](https://cloud.githubusercontent.com/assets/2483678/12572813/a44da89a-c3b6-11e5-9360-563d571d8ccc.png)

**Comment by nicolo-ribaudo** [comment]
This is a dupllicate of #2791 (fixed by #2792)

**Comment by jugglinmike** [comment]
@nicolo-ribaudo is correct--this has been fixed in `master`. The fix will be published in the next (post-2.9.1) version of JSHint.
